### PR TITLE
Fix manual CI workflow and add `name` option.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Signed-off-by: Yury-Fridlyand <yuryf@bitquilltech.com>

### Description
* Fix manual CI workflow.
* Add `name` optional parameter; If not given `<branch name> @ <commit sha>` is used.
 
### Issues Resolved
Workflow ignored all test errors and reported always success. Workflow was introduced in #837.
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Yury-Fridlyand** [comment]
image
image

File: .github/workflows/sql-test-workflow.yml [hunk 1]
```diff
@@ -2,6 +2,13 @@ name: SQL Plugin Tests
 
 on:
   workflow_dispatch:
+    inputs:
+      name:
```
Comment (acarbonetto) [inline .github/workflows/sql-test-workflow.yml:6]: please add a description, maybe with example? `<branch name> @ <commit sha>`

Comment (acarbonetto) [inline .github/workflows/sql-test-workflow.yml:6]: can we call this `branch_commit` or similar?

**Review by acarbonetto — APPROVED** [review]

File: .github/workflows/sql-test-workflow.yml (hunk 1 above)
Comment (Yury-Fridlyand) [reply .github/workflows/sql-test-workflow.yml:6]: That's a custom run name which could be specified to override the default value. Default is `<branch name> @ <commit sha>`

**Comment by penghuo** [comment]
> Workflow ignored all test errors and reported always success

What is the use case of the workflow?

**Comment by Yury-Fridlyand** [comment]
> What is the use case of the workflow?

It is supposed to run on dev/dbg branches to get results from all tests. This workflow could be triggered manually only.

**Review by penghuo — APPROVED** [review]

**Comment by penghuo** [comment]
Please do not backport to main which is not stable now.

**Review by MaxKsyunz — APPROVED** [review]

**Comment by Yury-Fridlyand** [comment]
Why backport always fail? Is something broken?

**Pull request merged**